worldedit-core/src/main/java/com/fastasyncworldedit/core/queue/implementation/QueueHandler.java

```diff
@@ -184,7 +183,7 @@ public <T> Future<T> syncWhenFree(Supplier<T> call) {
     }
 
     private <T> Future<T> sync(Runnable run, T value, Queue<FutureTask> queue) {
-        if (Fawe.isMainThread()) {
+        if (Fawe.isTickThread()) {
```

Comment:
This should be kept as isMainThread - it never makes sense on Folia to call this method.

That also applies to all other changes of that kind.

Reply:
Not really. Folia don't have a main thread. Only a global ticking region

Reply:
Yes, but calling the sync method does not make sense on Folia, and knowing that we are on a tick thread does not solve any issue

Reply:
I had some issues with other testing here. Thats is currently one solution that "works"

Reply:
If something works when using this method, it is either a) luck or b) doesn't actually need any synchronization.

Reply:
I think b is here. I tested with big areas(>1mio x 1mio of blocks)